static/app/utils/metrics/useBlockMetric.tsx

```diff
@@ -0,0 +1,76 @@
+import {addErrorMessage, addSuccessMessage} from 'sentry/actionCreators/indicator';
+import {t} from 'sentry/locale';
+import type {MetricMeta, MRI, Project} from 'sentry/types';
+import {getUseCaseFromMRI} from 'sentry/utils/metrics/mri';
+import {useMutation, useQueryClient} from 'sentry/utils/queryClient';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+
+export type BlockOperationType =
+  | 'blockMetric'
+  | 'unblockMetric'
+  | 'blockTags'
+  | 'unblockTags';
+
+type BlockMetricData = [MetricMeta[], unknown, unknown] | undefined;
+
+export const useBlockMetric = (project: Project) => {
+  const api = useApi();
+  const {slug} = useOrganization();
+  const queryClient = useQueryClient();
+
+  const options = {
+    mutationFn: (data: {
+      mri: MRI;
+      operationType: BlockOperationType;
+      tags?: string[];
```

Comment:
I would assume that we need to pass tags for a tag operation, right? Would be nice if the type could reflect that.